### PR TITLE
Use swift-lmdb instead of copying LMDB source

### PR DESCRIPTION
**Pull request by matthewbastien**
This is tracked by rdar://139431575 as a part of https://github.com/swiftlang/vscode-swift/issues/562

Adds swift-lmdb as a dependency and removes the copied lmdb source so that IndexStoreDB uses the same version of lmdb as swift-docc. This enables SourceKit-LSP to use swift-docc as a dependency to fulfill documentation requests on behalf of the VS Code Swift extension.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -18,7 +18,9 @@ let package = Package(
       name: "tibs",
       targets: ["tibs"])
   ],
-  dependencies: [],
+  dependencies: [
+    .package(url: "https://github.com/swiftlang/swift-lmdb.git", branch: "main"),
+  ],
```
Comment (ahoppen) [inline Package.swift:23]: Could you use a path based dependency if `SWIFTCI_USE_LOCAL_DEPS` is set? 

Similar to https://github.com/swiftlang/sourcekit-lsp/blob/c23b4ff776d34479ddc37d9d2133e0e7007ad35f/Package.swift#L571

Comment (matthewbastien) [reply Package.swift]: Done.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/swiftlang/swift/pull/77624

@swift-ci please test

**Comment by matthewbastien** [comment]
swiftlang/swift-lmdb#8

@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/swiftlang/swift/pull/77624
https://github.com/swiftlang/swift-lmdb/pull/8

@swift-ci please test Windows platform

**Review by bnbarham — APPROVED** [review]
LGTM - running a toolchain build in https://github.com/swiftlang/swift/pull/77624

**Pull request merged**